source/common/upstream/upstream_impl.cc

```diff
@@ -1432,16 +1430,18 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(const HostVector& new_hosts,
 
   // Remove hosts from current_priority_hosts that were matched to an existing host in the previous
   // loop.
-  for (auto itr = current_priority_hosts.begin(); itr != current_priority_hosts.end();) {
-    auto existing_itr = existing_hosts_for_current_priority.find((*itr)->address()->asString());
+  auto current_erase_from = current_priority_hosts.end();
+  for (auto i = current_priority_hosts.begin(); i != current_erase_from;) {
+    auto existing_itr = existing_hosts_for_current_priority.find((*i)->address()->asString());
 
     if (existing_itr != existing_hosts_for_current_priority.end()) {
       existing_hosts_for_current_priority.erase(existing_itr);
-      itr = current_priority_hosts.erase(itr);
+      *i = *(--current_erase_from);
     } else {
-      itr++;
+      i++;
```

Comment:
Please use ++i here. In general this is preferred for consistency, but for iterators there is a material difference when the impl doesn't need to return the previous value of the iter.